data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/processor/peerforwarder/PeerForwarder.java

```diff
@@ -112,10 +132,19 @@ private void processRequest(final TraceServiceGrpc.TraceServiceBlockingStub clie
                                 final ExportTraceServiceRequest request,
                                 final List<Record<ExportTraceServiceRequest>> localBuffer) {
         if (client != null) {
+            final String peerIp = client.getChannel().authority();
+            final Timer forwardRequestTimer = forwardRequestTimers.computeIfAbsent(
+                    peerIp, ip -> pluginMetrics.timer(String.format("%s:%s", FORWARD_REQUEST_LATENCY_PREFIX, ip)));
+            final Counter forwardedRequestCounter = forwardedRequestCounters.computeIfAbsent(
+                    peerIp, ip -> pluginMetrics.counter(String.format("%s:%s", FORWARD_REQUEST_SUCCESS_PREFIX, ip)));
+            final Counter forwardRequestErrorCounter = forwardRequestErrorCounters.computeIfAbsent(
+                    peerIp, ip -> pluginMetrics.counter(String.format("%s:%s", FORWARD_REQUEST_ERRORS_PREFIX, ip)));
             try {
-                client.export(request);
+                forwardRequestTimer.record(() -> client.export(request));
+                forwardedRequestCounter.increment();
```

Comment:
I'd be interested in knowing if we actually need both of these metrics. Could we potentially just count the number of timing metrics to derive the count (on the frontend UI, not in code).

Reply:
Good question! So in the results of timer, there is indeed a TOTAL number of calls. The issue is that it will count # of calls on client.export no matter it fails or not. Therefore we still need independent counters for success and failures.